File: SDKs/amazonaws.com/mturk-requester/2017-01-17/python/src/sdk/models/shared/createhittyperequest.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class CreateHitTypeRequest:
    assignment_duration_in_seconds: int = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('AssignmentDurationInSeconds') }})
    description: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Description') }})
    reward: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Reward') }})
    title: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Title') }})
    auto_approval_delay_in_seconds: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AutoApprovalDelayInSeconds') }})
    keywords: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Keywords') }})
    qualification_requirements: Optional[List[QualificationRequirement]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('QualificationRequirements') }})
    

File: SDKs/amazonaws.com/mturk-requester/2017-01-17/typescript/src/sdk/models/shared/createhittyperequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { QualificationRequirement } from "./qualificationrequirement";



export class CreateHitTypeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=AssignmentDurationInSeconds" })
  assignmentDurationInSeconds: number;

  @SpeakeasyMetadata({ data: "json, name=AutoApprovalDelayInSeconds" })
  autoApprovalDelayInSeconds?: number;

  @SpeakeasyMetadata({ data: "json, name=Description" })
  description: string;

  @SpeakeasyMetadata({ data: "json, name=Keywords" })
  keywords?: string;

  @SpeakeasyMetadata({ data: "json, name=QualificationRequirements", elemType: QualificationRequirement })
  qualificationRequirements?: QualificationRequirement[];

  @SpeakeasyMetadata({ data: "json, name=Reward" })
  reward: string;

  @SpeakeasyMetadata({ data: "json, name=Title" })
  title: string;
}
